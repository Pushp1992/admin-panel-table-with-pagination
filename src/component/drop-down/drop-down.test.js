import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import {mockdata} from './mock-data';
import DropDownField from './drop-down-field';
import '@testing-library/jest-dom';

afterEach(() => {
    cleanup();
});

it('matches snapshot with actual DOM tree', () => {
    const component = renderer.create(<DropDownField items={mockdata} />);
    const DOMTree = component.toJSON();
    expect(DOMTree).toMatchSnapshot();
});

it('renders dropdown component', () => {
    render(<DropDownField items={mockdata} />);
    const element = screen.getAllByTestId('dropdown');
    expect(element[0]).toBeInTheDocument();
});
