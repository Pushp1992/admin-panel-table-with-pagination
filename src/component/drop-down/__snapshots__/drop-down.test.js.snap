// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with actual DOM tree 1`] = `
<div
  className="DropDownWrapper-keMuLR"
  data-testid="dropdown"
>
  <select>
    <option
      value=""
    >
      Select Filter
    </option>
    <option
      value="name"
    >
      Name
    </option>
    <option
      value="email"
    >
      Email
    </option>
    <option
      value="role"
    >
      Role
    </option>
  </select>
</div>
`;
